### PR TITLE
Improves documentation for Jack-In ENV and debugging Java / JVM. 

### PR DESCRIPTION
**Pull request by ieugen**
## What has Changed?

Improves documentation for Jack-In and debugging Java / JVM. 
This started as debugging enhancement.
Feedback is welcomed. 

## My Calva PR Checklist
I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
~- [ ] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.~
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
~- [ ] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test.~
     ~- [ ] Tested the particular change~
     ~- [ ] Figured if the change might have some side effects and tested those as well.~
     ~- [ ] Smoke tested the extension as such.~
~- [ ] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.~
     ~- [ ] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)~
     ~- [ ] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.~
~- [ ] Created the issue I am fixing/addressing, if it was not present.~

Ping @pez, @bpringe

### CODE REVIEW

File: docs/site/debugger.md [hunk 1]
```diff
@@ -179,3 +179,47 @@ Most importantly, make sure you have `cider/cider-nrepl` as a dependency, and `c
 ```sh
 clojure -Sdeps '{:deps {nrepl/nrepl {:mvn/version,"0.8.3"},cider/cider-nrepl {:mvn/version,"0.25.8"}}}' -m nrepl.cmdline --middleware "[cider.nrepl/cider-middleware]"
 ```
+
+## Passing options to the REPL JVM
+
+There are times when clojure debugging tools are not enough or not right for the job.
+This is usually true when use an (open source) Java library and you want to set some brake points in Java code.
+For those cases and others, you need to start the JVM in debug mode.
+And for that you nee
```
Comment (bpringe) [inline docs/site/debugger.md]: I think this sentence got cut off somehow.

File: docs/site/debugger.md [hunk 2]
```diff
@@ -179,3 +179,47 @@ Most importantly, make sure you have `cider/cider-nrepl` as a dependency, and `c
...
+        "JAVA_TOOL_OPTIONS": "${env:JAVA_TOOL_OPTIONS} -Dorg.slf4j.simpleLogger.defaultLogLevel=TRACE -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=7896"
+    }
+}
+```
+
+Once you saved the file, the next time you `Jack in` the project, you should see something like:
+
+```shell
+clojure -Sdeps '{:deps {nrepl/nrepl {:mvn/version,"0.8.3"},cider/cider-nrepl {:mvn/version,"0.26.0"}}}' -A:debug -m nrepl.cmdline --middleware "[cider.nrepl/cider-middleware]"
+WARNING: Implicit use of clojure.main with options is deprecated, use -M
```
Comment (bpringe) [inline docs/site/debugger.md]: This warning will be removed in the next release, so we can remove it from here.

```suggestion
```

**Review by bpringe — CHANGES_REQUESTED** [review]
Thanks for improving the docs!

File: docs/site/debugger.md [hunk 3]
```diff
@@ -179,3 +179,47 @@ Most importantly, make sure you have `cider/cider-nrepl` as a dependency, and `c
...
+Tipical use cases:
+
+* Change java logger configuration for the REPL via java system properties: e.g `-Dorg.slf4j.simpleLogger.defaultLogLevel=TRACE`
+* Enable JVM debugger, change VM memory size, etc
+
+Calva supports passing environment variables via `jacInEnv`.
+You can set that option inside VSCode `settings.json` file.
+
+You can configre global `settings.json` file or a project wide version.
+I think it's best to put these options inside your project, inside `<project-root>/.vscode/settings.json`.
```
Comment (PEZ) [inline docs/site/debugger.md]: This depends on the options set. I don't think we need to recommend either.

File: docs/site/debugger.md [hunk 4]
```diff
@@ -179,3 +179,47 @@ Most importantly, make sure you have `cider/cider-nrepl` as a dependency, and `c
...
 ```
+
+## Passing options to the REPL JVM
+
+There are times when clojure debugging tools are not enough or not right for the job.
+This is usually true when use an (open source) Java library and you want to set some brake points in Java code.
+For those cases and others, you need to start the JVM in debug mode.
+And for that you nee
+
+Tipical use cases:
```
Comment (PEZ) [inline docs/site/debugger.md]:
```suggestion
Typical use cases:
```

**Comment by agoldhammer** [comment]
"brake points"  should read "break points"

File: docs/site/debugger.md (hunk 1 above)
Comment (ieugen) [reply docs/site/debugger.md]: thanks, removed.

File: docs/site/debugger.md (hunk 2 above)
Comment (ieugen) [reply docs/site/debugger.md]: I removed the warning.

File: docs/site/debugger.md (hunk 3 above)
Comment (ieugen) [reply docs/site/debugger.md]: I removed the recommendation.

**Comment by ieugen** [comment]
I resolved the review requests and also added a commit with the brakpoints -> break points change.
Seems to be correct https://www.merriam-webster.com/dictionary/break%20point . 
It can be easily reverted if necessary.

**Comment by PEZ** [comment]
Thanks!

I think it should be *breakpoint*, see https://en.wikipedia.org/wiki/Breakpoint

**Comment by ieugen** [comment]
Done.

**Review by PEZ — APPROVED** [review]
LGTM!

**Comment by PEZ** [comment]
Thanks again for improving Calva!

**Comment by ieugen** [comment]
Thank you for doing such great work. 
It is a pleasure working with Calva and interacting with you!

**Pull request merged**